src/main/java/com/yahoo/bullet/record/BulletRecord.java

```diff
@@ -125,6 +125,44 @@ public Object get(String field, int index) {
         return casted.get(index);
     }
 
+    /**
+     * Gets a field from the record by identifier.<br>
+     * <br>
+     * For example, suppose a record has a map of boolean maps called "aaa". Then <br>
+     * - "aaa" identifies that map of maps <br>
+     * - "aaa.bbb" identifies the inner map that "aaa" maps "bbb" to (if it exists) <br>
+     * - "aaa.bbb.ccc" identifies the boolean that "aaa.bbb" (if it exists) maps "ccc" to (if it exists) <br>
+     * <br>
+     * For a list element, the index is the key, e.g. "my_list.0" or "my_list.0.some_key"
+     *
+     * @param field The non-null identifier of the field to get.
+     * @return The value of the field or null if it does not exist.
+     */
+    @SuppressWarnings("unchecked")
+    public Object extract(String field) {
```

Comment:
change field to identifier

Reply:
Agree to the identifier rename.  

There are two approaches to this:

1. We can make this method protected and add add helper methods for ```Object get(String field, String subKey, String subSubKey)``` and ```Object get(String field, int index, String subKey)```, just like the other gets. This allows only those methods to be used. This makes the dot separated convention of defining fields not be defined in BulletRecord and just how we use it.

2. We can leave it as public interface but still add those other get methods since we already have the first two gets. If so, can we can call this ```extractField```? (to match [this method](https://github.com/bullet-db/bullet-core/blob/master/src/main/java/com/yahoo/bullet/common/Utilities.java#L103) in bullet-core which can now be removed)